pg-extend/src/pg_fdw.rs

```diff
@@ -418,8 +419,8 @@ impl<T: ForeignData> ForeignWrapper<T> {
     ) -> *mut pg_sys::TupleTableSlot {
         let wrapper = Box::from_raw((*rinfo).ri_FdwState as *mut Self);
 
-        let fields = Self::tts_to_hashmap(slot, &(*slot).tts_tupleDescriptor);
-        let fields_with_index = Self::tts_to_hashmap(plan_slot, &(*plan_slot).tts_tupleDescriptor);
+        let fields = Self::tts_to_hashmap(slot, (*slot).tts_tupleDescriptor);
```

Comment:
Do we have to be concerned with any of these parameters possibly being null?

Reply:
I don't think the tupleDescriptor will ever be null. I could add a check for it.

Reply:
There's a question of how defensive we want to be I think. Do we want to check every single pointer as null? that'd be safer but less performant

Reply:
I think it depends on what Postgres "guarantees" in the API. If it claims it will never be null, then I guess it's probably not valuable. This is an area, where I'm not always 100% clear on what the best thing to do when making "safe" APIs in Rust.

So really what I was asking was if PG claims these will never be null, we don't necessarily need to check each pointer.

Reply:
postgres docs usually mention if something is nullable, and none of these are listed